File: ClientAPI/src/main/java/pt/upskill/clientapi/DTOs/LoginDTO.java
package pt.upskill.clientapi.DTOs;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Represents a login data transfer object.
 * This class is used to hold the login credentials (email and password) of a user.
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class LoginDTO {

    /**
     * The email of the user.
     * This field is used to identify the user during the login process.
     */
    private String email;

    /**
     * The password of the user.
     * This field is used to authenticate the user during the login process.
     */
    private String password;

}

File: ForumAPI/ForumAPI/DTOs/LoginDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ForumAPI.DTOs
{
    public class LoginDTO
    {
        [Required]
        public string Email { get; set; }

        [Required]
        public string Password { get; set; }

        public LoginDTO(string Email, string Password)
        {
            this.Email = Email;
            this.Password = Password;
        }
    }
}
